tests/testthat/test-tcplhit2_core.R

```diff
@@ -13,3 +13,133 @@ test_that("tcplhit2 works", {
+  # Estimated BMD is below the upper 'threshold dose' and
+  # above the 'reference dose' (highest expt dose).
+  output_after_ten <- tcplhit2_core(params, conc, resp, cutoff, onesd,
+                                bmed=0, conthits=T, aicc=F, bmd_up_bnd = 10)
+
+  # checks for bmd estimates
+  expect_equal(output_before$bmd, 299.9274, tolerance = 1e-3)
+  expect_equal(output_after$bmd, 160, tolerance = 1e-3)
+  expect_equal(output_after_ten$bmd, 299.9274, tolerance = 1e-3)
+
+  # checks for bmd confident lower bound
+  expect_equal(output_before$bmdl, 217.343, tolerance = 1e-3)
+  expect_equal(output_after$bmdl, 77.41564, tolerance = 1e-3)
+  expect_equal(output_after_ten$bmdl, 217.343, tolerance = 1e-3)
+
+  # checks for bmd confident upper bound
+  expect_equal(output_before$bmdu, 475.0638, tolerance = 1e-3)
+  expect_equal(output_after$bmdu, 335.1364, tolerance = 1e-3)
+  expect_equal(output_after_ten$bmdu, 475.0638, tolerance = 1e-3)
+})
+
+test_that("tcplhit2 BMD lower boundary censor checks", {
+  data("mc3")
+
+  temp <- mc3[mc3$logc<= -2,"resp"]
+  bmad <- mad(temp)
+  onesd <- sd(temp)
+  cutoff <- 3*bmad
+
+  # load example data from mc3
+  spid <- unique(mc3$spid)[94]
+  ex_df <- mc3[is.element(mc3$spid,spid),]
+  conc <- 10**ex_df$logc # back-transforming concentrations on log10 scale
+  resp <- ex_df$resp
+  conc2 <- conc[conc>0.41]
+  resp2 <- resp[which(conc>0.41)]
+
+  params <- tcplfit2_core(conc2, resp2, cutoff,
+                          force.fit = T, bidirectional = F)
+
+  # hit-calling with and without using the BMD boundary argument
+  output_before <- tcplhit2_core(params, conc2, resp2, cutoff, onesd,
+                                 bmed=0, conthits=T, aicc=F)
+  # Estimated BMD is less than the lower 'threshold dose'.
+  output_after <- tcplhit2_core(params, conc2, resp2, cutoff, onesd,
+                                bmed=0, conthits=T, aicc=F, bmd_low_bnd = 0.8)
+  # Estimated BMD is above the lower 'threshold dose' and
+  # below the 'reference dose' (lowest expt dose).
+  output_after_ten <- tcplhit2_core(params, conc2, resp2, cutoff, onesd,
+                                bmed=0, conthits=T, aicc=F, bmd_low_bnd = 0.1)
+
+  # checks for bmd estimates
+  expect_equal(output_before$bmd, 0.302, tolerance = 1e-3)
+  expect_equal(output_after$bmd, 0.48, tolerance = 1e-3)
+  expect_equal(output_after_ten$bmd, 0.302, tolerance = 1e-3)
+
+  # checks for bmd confident lower bound
+  expect_equal(output_before$bmdl, 0.1048, tolerance = 1e-3)
+  expect_equal(output_after$bmdl, 0.2826, tolerance = 1e-3)
+  expect_equal(output_after_ten$bmdl, 0.1048, tolerance = 1e-3)
+
+  # checks for bmd confident upper bound
+  expect_equal(output_before$bmdu, 0.815, tolerance = 1e-3)
+  expect_equal(output_after$bmdu, 0.9928, tolerance = 1e-3)
+  expect_equal(output_after_ten$bmdu, 0.815, tolerance = 1e-3)
+})
+
+test_that("tcplhit2 BMD boundary censor checks", {
```

Comment:
Suggest adding similar test but add the lower and upper boundary that are 0.1 and 10.

Reply:
This tests I would suggest updating to "tcplhit2 BMD boundary check - no censoring needed".  Then the other one (suggested above) "tcplhit2 BMD boundary check - censoring required".  Or something to that effect.

Let me know if you have further questions on this.